tests/atoms/gaia_data_layer.js

```diff
@@ -43,46 +43,49 @@ var GaiaDataLayer = {
     marionetteScriptFinished(true);
   },
 
-  disableBluetooth: function() {
+  setBluetooth: function(aValue) {
     var bluetooth = window.navigator.mozBluetooth;
-    if (bluetooth.enabled) {
-      console.log('trying to disable bluetooth');
-      this.setSetting('bluetooth.enabled', false, false);
-      waitFor(
-        function() {
-          marionetteScriptFinished(true);
-        },
-        function() {
-          console.log('bluetooth enable status: ' + bluetooth.enabled);
-          return bluetooth.enabled === false;
-        }
-      );
-    }
-    else {
-      console.log('bluetooth already disabled');
+    var adapter = bluetooth.getAdapters()[0];
+
+    if (adapter.state == aValue) {
+      console.log('bluetooth already ' + aValue);
       marionetteScriptFinished(true);
+      return;
     }
+
+    waitFor(
+      function() {
+        console.log('bluetooth ' + aValue);
+        marionetteScriptFinished(true);
+      },
+      function() {
+        console.log('bluetooth enable status: ' + bluetooth.getAdapters()[0].state);
+
+        // Wait for the adapter state to become 'disabled' or 'enabled'
+        if (adapter.state == 'disabling' || adapter.state == 'enabling')
+          return false;
+
+        var isnotValue = aValue == 'enabled' ? 'disabled' : 'enabled';
```

Comment:
sorry, I'm not familiar with c style conditional operator on python. is this same as below?
if aValue == 'enabled:
   isnotValue = 'disabled'
else
   isnotValue = 'enabled'

if so, can we just check for (adapter.state != aValue) below, instead of checking (adapter.state == isnotValue) ?